controller/hydrator_dependencies.go

```diff
@@ -51,6 +53,19 @@ func (ctrl *ApplicationController) GetRepoObjs(origApp *appv1.Application, drySo
 
 	// FIXME: use cache and revision cache
 	objs, resp, _, err := ctrl.appStateManager.GetRepoObjs(app, drySources, appLabelKey, dryRevisions, true, true, false, project, false)
+	for _, obj := range objs {
+		// Remove Argo CD's internal tracking annotation before committing to Git.
+		// This annotation is used only for in-memory tracking and should not be persisted.
+		annotations := obj.GetAnnotations()
+		if annotations != nil {
+			delete(annotations, common.AnnotationKeyAppInstance)
+			if len(annotations) == 0 {
+				obj.SetAnnotations(nil)
+			} else {
+				obj.SetAnnotations(annotations)
+			}
+		}
+	}
```

Comment:
Switch this to a call to `RemoveAppInstance`?

Reply:
GetRepoObjs in hydrator_dependencies.go has access to the settings manager. I think it can pull the tracking method from there.

Reply:
Pulling the tracking method is possible, something like `ctrl.settingMgr.GetTrackingMethod()`, however, we still can't call the `RemoveAppInstance` there.